src/vm/threads.cpp

```diff
@@ -908,11 +910,12 @@ void DestroyThread(Thread *th)
 #ifdef FEATURE_PERFTRACING
     // Before the thread dies, mark its buffers as no longer owned
     // so that they can be cleaned up after the thread dies.
-    EventPipeBufferList *pBufferList = th->GetEventPipeBufferList();
+    EventPipeBufferList *pBufferList = GetThreadEventBufferList();
     if(pBufferList != NULL)
```

Comment:
Same issue as mentioned below. If the buffers aren't owned by Thread then Thread related code shouldn't be responsible for cleaning them up.